go-controller/pkg/ovn/obj_retry.go

```diff
@@ -126,3 +136,447 @@ func (r *retryObjs) getObjRetryEntry(key string) *retryObjEntry {
 	}
 	return nil
 }
+
+func splitNamespacedName(namespacedName string) (string, string) {
+	sep := "/"
+	if strings.Contains(namespacedName, sep) {
+		s := strings.SplitN(namespacedName, sep, 1)
+		if len(s) == 2 {
+			return s[0], s[1]
+		}
+	}
+	return namespacedName, ""
+
+}
+
+func areResourceObjectsEqual(objType reflect.Type, obj1, obj2 interface{}) (bool, error) {
```

Comment:
so I guess here we would put in resource specific comparisons to know if something is equal? For example we may care about some annotation on a node and not a network policy to determine if an object changed?

Reply:
yeah, that's the idea. 

The version I pushed on Friday didn't work too well with nodes because it didnt distinguish between resource types where an update means delete+add and those where we actually have an update function. Now it seems to work :) I will apply it on top of the latest changes from mohamed's PR.